### DISCUSSION TITLE
Add support for V380E2_CA HwV380E12_WF3_PTZ_20160217

### DISCUSSION THREAD

**Discussion by 0zoon0** [Arkady23/V380-Web#31] [Ideas]
Hello,
I really appreciate work you do here.

Could you add support for this camera please?
Latest firmware version recently updated to: V380E2_CA_V2.5.10.6
Hardware version after update: HwV380E12_WF3_PTZ_20160217
App version after update: AppV380E12_PTZ_V2.5.10.6_20190803

I unzipped V380E2_CA_V2.5.10.6-20200106.zip and can see V380E2_CA in the very first line in .patch file.

I tried httpd_V380E2_CA_v20210612.zip but it didn't work. The camera sees there is an update, but could not access it by http (and port 80 is not opened). I guess that because the Hardware versions are different: V380E12 vs V380E13.

What is your thought on this @Arkady23 ?

Thank you.

**Comment by Arkady23** [comment]
Do you have an official patch? Can you post it?

**Comment by 0zoon0** [comment]
[V380E2_CA_V2.5.10.6-20200106.zip](https://github.com/Arkady23/V380-Web/files/8046978/V380E2_CA_V2.5.10.6-20200106.zip)
Here it is.

Thank you.

**Comment by Arkady23** [comment]
This patch is for version V380E2_CA. I didn't understand anything from what you wrote above. If http was working for you, but stopped after updating the camera, then this is normal. It can be. If this is the case, then you need [to install http again](https://github.com/Arkady23/V380-Web/releases/download/20210530/httpd_V380E2_CA_v20210612.zip). This link is to your patch httpd_V380E2_CA_v20210612.zip. If the patch is installed, then http can still be updated to the latest version using the web interface.

**Comment by 0zoon0** [comment]
Sorry, will try to be more explicit:
"If http was working for you" - no, it never worked.
"but stopped after updating the camera" - no, sorry, I updated my camera to the latest official version, but this has nothing to do with your work. I should not have mentioned that.
"then you need to install http again. This link is to your patch httpd_V380E2_CA_v20210612.zip" - as I said, I installed and it didn't work.
I guess that's because httpd_V380E2_CA_v20210612.zip is for HwV380E13_WF3_PTZ but my camera is HwV380E12_WF3_PTZ_20160217 (E13 vs E12 - could that be the problem?)

Sorry for confusing you from the beginning. Let me use google translate to say it again:
У меня есть эта камера: HwV380E12_WF3_PTZ_20160217
Я пробовал httpd_V380E2_CA_v20210612.zip, но это не сработало. Устанавливаю файлы на сд карту, думаю камера увидела, что есть какие-то обновления, но при открытии в браузере http://192.168.0.90 страница не открылась.

Большое спасибо.

**Comment by Arkady23** [comment]
For me, which particular camera E13 or E12 does not play any role. Only the camera's Hwname plays a role. Have you given the patch that you have installed now? Has any other patch been installed after that? If it is the latter, then its version is 2051006. Then I can make the version 1 larger for the experiment, for example, 2051007. So that the patch is guaranteed to be installed. I also want to know. After trying to install http, you have next files on your sd
local_update.conf
update_record
patch_reuse
and the updatepatch folder?

**Comment by Arkady23** [comment]
Let's try this patch
[httpd_V380E2_CA_9090909_v20210620.zip](https://github.com/Arkady23/V380-Web/files/8047657/httpd_V380E2_CA_9090909_v20210620.zip)
I have brought my patch closer to the logic of the official patch. Maybe it will work out with this?

**Comment by 0zoon0** [comment]
У меня была другая более старая версия, но я обновил ее до V380E2_CA_V2.5.10.6-20200106.zip. Только после этого обновления я попробовал ваш хак. Никаких других обновлений, исправлений или хаков не устанавливалось.
После установки httpd_V380E2_CA_v20210612.zip и запуска/ожидания/остановки камеры все эти четыре файла все еще находились на SD-карте (HTTP не работал).
А вот этот патч httpd_V380E2_CA_9090909_v20210620.zip сработал хорошо. Я могу получить доступ к странице HTTP, я вижу, как создаются папки для записи, RTSP также работает, но ссылка ONVIF не работает (но я думаю, это потому, что прошивка камеры не поддерживает /onvif/device_service).

Большой лайк и спасибо за быстрый ответ и труд!!!

I had some other older version, but I updated it to V380E2_CA_V2.5.10.6-20200106.zip. Just after this update I tried your hack. No other updates or patches or haks were installed.
After httpd_V380E2_CA_v20210612.zip installation and camera start/wait/stop all those four files still were on SD card (HTTP not working). 
But this patch httpd_V380E2_CA_9090909_v20210620.zip worked well. I can access HTTP page, I can see recording folders being created, RTSP works as well, but ONVIF link is not working (but I guess this is because camera firmware is not supporting /onvif/device_service).

Big like and thank you for your quick reply and hard work!!!

**Reply by keyj9** [reply]
how do you install it? you copy the files to the sdcard telnet into the camera and then what?

**Comment by ariefridwan888-creator** [comment]
firmware download HwV380E12_WF3_PTZ_20160217 please
